client/src/schema/config.ts

```diff
@@ -0,0 +1,71 @@
+import { getBranchDetails } from '../api';
+import { FetcherExtraProps } from '../api/fetcher';
+import { getEnvVariable, getGitBranch } from '../util/environment';
+import { isObject } from '../util/lang';
+
+const envBranchNames = [
+  'XATA_BRANCH',
+  'VERCEL_GIT_COMMIT_REF', // Vercel
+  'CF_PAGES_BRANCH', // Cloudflare Pages
+  'BRANCH' // Netlify. Putting it the last one because it is more ambiguous
+];
+
+export async function getBranch(fetchProps: Omit<FetcherExtraProps, 'workspacesApiUrl'>): Promise<string | undefined> {
+  try {
+    const env = getBranchByEnvVariable();
+    if (env) return env;
+  } catch (err) {
+    // Ignore
+  }
```

Comment:
[nitpick] Would probably move the try/catch to ``getBranchByEnvVariable``. It's not clear why we should catch, so easy to remove during a refactor.

Reply:
Makes sense!